numpy/core/src/multiarray/methods.h

```diff
@@ -6,4 +6,37 @@ extern NPY_NO_EXPORT PyMethodDef array_methods[];
 NPY_NO_EXPORT const char *
 npy_casting_to_string(NPY_CASTING casting);
 
+/*
+ * Pathlib support.
+ * For Python >= 3.6, use the os.Pathlike interface.
+ * Else, for Python >= 3.4, use file = str(file) if file is a PurePath
+ * For older Python, do nothing.
+ */
+static inline PyObject *
+NpyPath_PathlikeToFspath(PyObject *file)
```

Comment:
I wonder if it would be better to leverage `numpy.compat.os_fspath` and `numpy.compat.os_PathLike` here. That would look something like the followin
```C
static PyObject *os_PathLike = NULL;
static PyObject *os_fspath = NULL;

npy_cache_import("numpy.compat", "os_PathLike", &os_PathLike);
if (os_PathLike == NULL) {
    return NULL;
}
npy_cache_import("numpy.compat", "os_fspath", &os_fspath);
if (os_fspath == NULL) {
    return NULL;
}

if (!PyObject_IsSubclass(file, os_PathLike)) {
    // assume this is a file handle or a string
    return file;
}
return PyObject_CallFunctionObjArgs(os_fspath, file, NULL);
```

Reply:
That seems reasonable. I was not aware of numpy.compat :)